src/Executables/Examples/SingletonTovProfile/SingletonTovProfile.hpp

```diff
@@ -0,0 +1,207 @@
+  using type = double;
+  static constexpr OptionString help{"The central mass density rho_c"};
+};
+
+/*!
+ *  The points are linearly spaced in the areal radius. For
+ *  reconstruction purposes, the barycentric rational interpolant to the
+ *  log_enthalpy-radius curve is the most well-behaved, and
+ *  is best interpolated using an order 3 barycentric rational interpolant.
+ *  Using an interpolant with 25 points results in a log enthalpy interpolant
+ *  that has a relative error of at most 1.e-8 at all radii, relative to the
+ *  numerically found result.
+ */
+struct NumberOfPoints {
+  using type = size_t;
+  static constexpr OptionString help{"The number of data points to output"};
+};
+}  // namespace TovSingletonCacheTags
+
+namespace Actions {
+struct SolveTovProfile {
+  template <typename DbTags, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static void apply(db::DataBox<DbTags>& /*box*/,
+                    tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& cache,
+                    const ArrayIndex& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) {
+    const std::string filename =
+        Parallel::get<TovSingletonCacheTags::Name>(cache);
+    const double central_mass_density =
+        Parallel::get<TovSingletonCacheTags::CentralMassDensity>(cache);
+    const size_t num_pts =
+        Parallel::get<TovSingletonCacheTags::NumberOfPoints>(cache);
+    const auto& equation_of_state =
+        Parallel::get<OptionTags::EquationOfState<true, 1>>(cache);
+
+    // Integrate TOV equations out to the surface of the star:
+    const double surface_log_enthalpy = 0.0;
+    const gr::Solutions::TovSolution tov_star(
+        equation_of_state, central_mass_density, surface_log_enthalpy);
+    const double outer_radius = tov_star.outer_radius();
+
+    // Define helper closures:
+    const auto pressure_from_density = [&equation_of_state](
+        const double density) noexcept {
+      return equation_of_state.pressure_from_density(Scalar<double>{density})
+          .get();
+    };
+    const auto energy_from_density = [&equation_of_state](
+        const double density) noexcept {
+      return equation_of_state
+          .specific_internal_energy_from_density(Scalar<double>{density})
+          .get();
+    };
+    const auto sound_speed_squared_from_density = [&equation_of_state](
+        const double density) noexcept {
+      return equation_of_state.chi_from_density(Scalar<double>{density}).get() +
+             equation_of_state
+                 .kappa_times_p_over_rho_squared_from_density(
+                     Scalar<double>{density})
+                 .get();
+    };
+
+    // Detailed output is written to the H5 file:
+    const std::string h5_file_name = filename + ".h5";
```

Comment:
I think this should either:
- Use the Observers infra
- Not delete the file if it exists, but instead error. I'm sure I'm not the only one who has accidentally overwritten data :) A failed execution is better than deleting data. Alternatively, you could do something like append a number to the file name (before the extension) and keep incrementing that number until you are able to write the file. Here the danger is many files and full disks.